pkg/api/converterfromapi.go

```diff
@@ -183,9 +193,18 @@ func convertPropertiesToV20160330(api *Properties, p *v20160330.Properties) {
 	}
 	p.AgentPoolProfiles = []*v20160330.AgentPoolProfile{}
 	for _, apiProfile := range api.AgentPoolProfiles {
-		v20160330Profile := &v20160330.AgentPoolProfile{}
-		convertAgentPoolProfileToV20160330(apiProfile, v20160330Profile)
-		p.AgentPoolProfiles = append(p.AgentPoolProfiles, v20160330Profile)
+		// We added a pool with a "_public" suffix when converting to API model;
+	    // we don't want to include that when converting back to a version-specific model
+		matched, err := regexp.MatchString(publicAgentPoolSuffix + "$", apiProfile.Name)
+		if !matched && err == nil {
+			v20160330Profile := &v20160330.AgentPoolProfile{}
+			convertAgentPoolProfileToV20160330(apiProfile, v20160330Profile)
+			p.AgentPoolProfiles = append(p.AgentPoolProfiles, v20160330Profile)
+		}
+	}
+	// Assign DNS Prefix to private agent pool from public agent pool
+	if len(p.AgentPoolProfiles) > 0 {
+		p.AgentPoolProfiles[0].DNSPrefix = api.AgentPoolProfiles[1].DNSPrefix	
```

Comment:
mostly looking good. Anyway we can use the above loop to inform what indexes to use here for this replacement?